rm/rm-server/src/test/java/functionaltests/nodesrecovery/RecoverLocalInfrastructureTest.java

```diff
@@ -0,0 +1,159 @@
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.objectweb.proactive.core.util.wrapper.BooleanWrapper;
+import org.ow2.proactive.resourcemanager.common.event.RMEventType;
+import org.ow2.proactive.resourcemanager.common.event.RMNodeSourceEvent;
+import org.ow2.proactive.resourcemanager.core.properties.PAResourceManagerProperties;
+import org.ow2.proactive.resourcemanager.frontend.ResourceManager;
+
+import functionaltests.monitor.RMMonitorEventReceiver;
+import functionaltests.utils.NodesRecoveryProcessHelper;
+import functionaltests.utils.RMFunctionalTest;
+import functionaltests.utils.RMTHelper;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 22/06/17
+ */
+public class RecoverLocalInfrastructureTest extends RMFunctionalTest {
+
+    private static final String START_CONFIG = "/functionaltests/config/functionalTRMProperties-RM-restart-clean-db.ini";
+
+    private static final String RESTART_CONFIG = "/functionaltests/config/functionalTRMProperties-RM-restart-keep-db.ini";
+
+    private static final String NODE_SOURCE_NAME = "LocalNodeSource" +
+                                                   RecoverLocalInfrastructureTest.class.getSimpleName();
+
+    private static final int NODE_NUMBER = 3;
+
+    private ResourceManager resourceManager = null;
+
+    @Before
+    public void prepareRM() throws Exception {
+        startRmAndCheckInitialState();
+    }
+
+    @After
+    public void killNodes() throws Exception {
+        // kill the remaining nodes that were preserved for the test
+        try {
+            NodesRecoveryProcessHelper.findRmPidAndSendSigKill("RMNodeStarter");
+        } catch (Exception e) {
+            // we know that doing this will cause exceptions, keep silent after the test
+        }
+    }
+
+    @Test
+    public void testRecoverLocalInfrastructureWithAliveNodes() throws Exception {
+        // kill the RM only by sending a SIGKILL and leave node processes alive
+        NodesRecoveryProcessHelper.findRmPidAndSendSigKill("RMStarterForFunctionalTest");
+        // nodes should be re-taken into account by the restarted RM
+        restartRmAndCheckFinalState(false);
+    }
+
+    @Test
+    public void testRecoverLocalInfrastructureWithDownNodes() throws Exception {
+        // kill the RM and also kill all node processes
+        rmHelper.killRM();
+        // nodes should be re-deployed by the restarted RM
+        restartRmAndCheckFinalState(true);
+    }
+
+    private void startRmWithConfig(String configurationFilePath) throws Exception {
+        String rmconf = new File(RMTHelper.class.getResource(configurationFilePath).toURI()).getAbsolutePath();
+        rmHelper.startRM(rmconf);
+        resourceManager = rmHelper.getResourceManager();
+    }
+
+    private void startRmAndCheckInitialState() throws Exception {
+        // start RM
+        startRmWithConfig(START_CONFIG);
+        Assert.assertTrue("Nodes preservation should be activated on RM start for the test",
```

Comment:
Replace Assert.assertTrue by assertThat so you'll keep using the same library in all the class